src/codegen/expression.rs

```diff
@@ -1369,11 +1375,22 @@ fn require(
         Target::Solana | Target::Substrate { .. } => {
             if opt.log_runtime_errors {
                 if let Some(expr) = expr {
-                    let error_string =
-                        error_msg_with_loc(ns, " require condition failed", Some(expr.loc()));
+                    let prefix = "runtime_error: ".as_bytes();
+                    let error_string = format!(
+                        " require condition failed in {},\n",
+                        ns.loc_to_string(false, &expr.loc())
+                    );
                     let print_expr = Expression::FormatString(
                         Loc::Codegen,
                         vec![
+                            (
+                                FormatArg::StringLiteral,
+                                Expression::BytesLiteral(
+                                    Loc::Codegen,
+                                    Type::Bytes(prefix.len() as u8),
+                                    prefix.to_vec(),
```

Comment:
Here you should create a new string which is `prefix` and `error_string` concatenated, and then create a single `FormatArg::StringLiteral` rather than two.

Reply:
I cannot understand why that should be done to be honest...
`StringConcat` calls `stdlib/ void concat()` which allocates a new buffer for the concatenated string and makes two copies, then another copy for the format. `FormatString` would allocate a buffer and do three copies total.
Is this concerning readability?

Reply:
I want to reach something like this: `format ("runtime_error: {require_expression} require condition failed,"). E.g wrapping the the require expression with a prefix and delimiter.
Why should I concatenate the prefix and the error string?

Reply:
given `require (1>2, "SESA!")`
I want to reach: `runtime_error: SESA! require condition failed,`

Reply:
Sorry, reading the code again I was confused.